dask/array/routines.py

```diff
@@ -639,29 +639,42 @@ def bincount(x, weights=None, minlength=0, split_every=None):
         if weights.chunks != x.chunks:
             raise ValueError("Chunks of input array x and weights must match.")
 
+    axis = (0,)
     token = tokenize(x, weights, minlength)
     args = [x, "i"]
     if weights is not None:
-        meta = np.bincount([1], weights=[1])
+        meta = np.array(np.bincount([1], weights=[1]), like=x._meta, dtype="int64")
```

Comment:
Please use [`array_safe`](https://github.com/dask/dask/blob/a1187b13321d69565b9c21359d739c239bd04c65/dask/array/utils.py#L445-L458), this will resolve issues with older versions of NumPy.

Reply:
Ah, thank you!